src/Parser/FrostMage/Modules/Items/ShardofExodar.js

```diff
@@ -0,0 +1,32 @@
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import { formatNumber } from 'common/format';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class ShardOfExodar extends Module {
+	static dependencies = {
+		combatants: Combatants,
+	};
+
+	LustDamage = 0;
+	
+  on_initialized() {
+    this.active = this.combatants.selected.hasFinger(ITEMS.SHARD_OF_EXODAR.id);
+  }
+
+  on_byPlayer_damage(event){
+	if (this.combatants.selected.hasBuff(SPELLS.HEROISM.id) || this.combatants.selected.hasBuff(SPELLS.BLOODLUST.id) || this.combatants.selected.hasBuff(SPELLS.TIME_WARP.id) || this.combatants.selected.hasBuff(SPELLS.NETHERWINDS.id) || this.combatants.selected.hasBuff(SPELLS.ANCIENT_HYSTERIA.id) || this.combatants.selected.hasBuff(SPELLS.DRUMS_OF_FURY.id) || this.combatants.selected.hasBuff(SPELLS.DRUMS_OF_RAGE.id) || this.combatants.selected.hasBuff(SPELLS.DRUMS_OF_THE_MOUNTAIN.id)) {
```

Comment:
A better solution would be to create an array with all the buffs, say
```jsx
const BLOODLUST_BUFFS = [
SPELLS.HEROISM.id,
SPELLS.BLOODLUST.id,
SPELLS.TIME_WARP.id,
....
];
```
and then in this event use `if (BLOODLUST_BUFFS.some(buff => this.combatants.selected.hasBuff(buff, event.timestamp)))`

Reply:
Yea i tried for about 2 hours to use an array instead and never could get it to work so i opted for listing them all out until i could figure out what i was doing wrong. But ill change it to what you suggested.